packages/svelte/src/compiler/phases/2-analyze/validation.js

```diff
@@ -66,12 +77,24 @@ function validate_element(node, context) {
 			}
 
 			if (attribute.name.startsWith('on') && attribute.name.length > 2) {
-				if (
-					attribute.value === true ||
-					is_text_attribute(attribute) ||
-					attribute.value.length > 1
-				) {
+				if (!is_expression_attribute(attribute)) {
 					error(attribute, 'invalid-event-attribute-value');
+				} else {
+					const value = attribute.value[0].expression;
+					if (
+						value.type === 'Identifier' &&
+						value.name === attribute.name &&
+						!context.state.scope.get(value.name) &&
+						global_events.has(attribute.name)
```

Comment:
We did that in Svelte 4 but we had to keep a list of globals that _are_ available, which always meant to play catch-up with new globals, and would lead to false positives in other cases where people have a global with that name declared.

Reply:
I mean purely in the context of event handlers, as here, since we're trying to deal with the `<button {onclick}>` case

Reply:
Oh I see - so the "is part of the list of events" check is removed and the rest stays the same? Yeah we can do that